source/fundamentals/crud/read-operations/text.txt

```diff
@@ -0,0 +1,284 @@
+
+.. code-block:: json
+
+   { "_id": 1, "title": "2 Fast 2 Furious ", "tags": ["undercover", "drug dealer"] } 
+   { "_id": 2, "title": "Fast 5", "tags": ["bank robbery", "full team"] }
+   { "_id": 3, "title": "Furious 7", "tags": ["emotional"] }
+   { "_id": 4, "title": "The Fate of the Furious", "tags": ["betrayal"] }
+
+Text Index
+----------
+
+You must create a **text index** before running a text search. A text
+index specifies the string or string array field for the query to run a
+text search on. A collection can only contain one text index.
+
+In the following examples, you run text searches on the ``title``
+field in the ``fast_and_furious_movies`` collection. To enable text
+searches on the ``title`` field, create a text index using the
+:ref:`Indexes <index-text-indexes>` builder with the following 
+snippet: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/SearchText.java
+   :language: java
+   :dedent:
+   :start-after: begin textIndex
+   :end-before: end textIndex
+
+.. note::
+
+   If you want to run text searches on both the ``title`` and
+   ``tags`` field in the ``fast_and_furious_movies`` collection, you
+   need to create an text index to include them.
+
+   If you already have an existing text index, you need to drop the
+   index by specifying the existing text index field name with the
+   following snippet: 
+
+   .. code-block:: java
+
+      collection.dropIndex("field_name");
+   
+   If you want to delete all existing indexes except ``_id``, specify "*". 
+
+   Then, you can create a :ref:`compound index <index-compound-indexes>`. A
+   compound index is a single index that holds references to multiple
+   fields within a collection's documents. This allows you to specify
+   multiple text fields using one text index.
+
+   To enable text searches on the ``title`` and ``tag`` fields, create a
+   compound index using the following snippet:  
+
+   .. code-block:: java
+
+      collection.createIndex(Indexes.compoundIndex(Indexes.text("title"), Indexes.text("tags")));
+
+Text Search
+-----------
+
+Use the ``Filters.text()`` method to specify a text search. 
+
+The ``Filters.text()`` method uses the :doc:`Filters builder
+</fundamentals/builders/filters>` to create an instance of a :ref:`Bson
+<bson>` type. The instance is a query filter that specifies how the text
+search should operate, which you pass to the ``find()`` method. 
+
+When the ``find()`` method executes, MongoDB runs a text search on the
+fields indexed with the text index on the collection. The text search
+returns documents that contain one or more of the search terms and a
+relevance score for each result. For more information on relevance
+scores, see the :ref:`Sort by Relevance <sort_by_relevance>` section. 
+
+You can include ``TextSearchOptions`` as the second parameter of the
+``Filters.text()`` method to specify text search options such as case
+sensitivity. By default, text searches run without case sensitivity
```

Comment:
I think you're missing a description of what TextSearchOptions is in this revision. I thought you had a sentence about it before.

Reply:
It does say what it is in the first sentence. What are you looking for to be in the description?

Reply:
Sorry, I must have been looking at an older version.